CoreScriptsRoot/Modules/DeveloperConsoleModule.lua

```diff
@@ -2719,12 +2719,16 @@ end
 -- Permissions --
 -----------------
 do
-	local permissions;
+	local permissionsLoading, permissions = false;
 	function DeveloperConsole.GetPermissions()
+		while permissionsLoading do wait() end
```

Comment:
Even if we spawn a new thread, I'm pretty sure we'd need to still keep the debounce, as everytime you attempt to open the console, it's going to check permissions. If we remove the debounce it's going to keep calling all of that api stuff everytime it wants to open until it goes through and creates permissions, which i don't think is what we want to have happen.